### ISSUE TITLE
How to implement findemement common method with pywinauto?

### ISSUE THREAD

**Issue by tibile** [pywinauto/pywinauto#479]
As you know, on selenium, we can find element of web by using By method. But on pywinauto, I don't how can do that. Could you please help me to do that. Thanks

**Comment by vasily-v-ryabov** [comment]
Method `child_window(...)` does almost the same, but you can call the actual search later by `.wrapper_object()` or by calling an action like `.click_input()`.

See [Window Specification](https://pywinauto.readthedocs.io/en/latest/getting_started.html#window-specification) chapter. This is what created before `wrapper_object()`.

Example:
```python
OK_btn = app.window(title_re=".*Main Title").child_window(control_type="Button", title="OK").wrapper_object() # actual button is found

# actual button may not exist at the moment
OK_btn_spec = app.window(title_re=".*Main Title").child_window(control_type="Button", title="OK")

# calling an action will find the actual button automatically (wrapper_object() is called implicitly)
app.window(title_re=".*Main Title").child_window(control_type="Button", title="OK").click_input()
```

More detailed list of criteria for `child_window(...)` can be one of those:
https://pywinauto.readthedocs.io/en/latest/code/pywinauto.findwindows.html#pywinauto.findwindows.find_elements

**Comment by tibile** [comment]
Thanks @vasily-v-ryabov ,
I have one more question about click function:
Where is the different between {control}.click() and {control}.click_input().
Thank you a lot!
Binh Le

**Comment by vasily-v-ryabov** [comment]
`click()` is an alias for `InvokePattern` from MS UI Automation API. Usually its meaning is the same as click action but without actual mouse cursor moving. If this pattern is not well implemented on the app side, we can't guarantee if it works. Method `click_input()` does move the mouse and performs realistic click like an end user.

This message ("An event was unable to invoke any of the subscribers") could be produced by an element that doesn't exist any more if I remember correctly.

**Comment by tibile** [comment]
Yes, I tried to use the click_input() method and it worked correctly as you suggest.
But one more thing, when I try to find an element, the test script run too slow if the screen name is specific.
For example: 
If I find an element on ParkingSoft page, the action performs too fast. But if we find in a form like: parkingsoft.Timeform, the performance of script run too slow. So do you know what is the root cause of it? 
Thanks.

**Comment by vasily-v-ryabov** [comment]
It depends on number of elements on the form. Usually adding `class_name` or `control_type` to `child_window(...)` speeds it up.

**Comment by tibile** [comment]
amazing, thank you for suggestion, love it

**Comment by vasily-v-ryabov** [comment]
@tibile you're welcome! If you love what we do, just star the repo to boost its [rating](https://github.com/pywinauto/pywinauto/wiki/UI-Automation-tools-ratings).

**Comment by tibile** [comment]
sure, I rated for it. <3
